src/post_compute/signer.rs

```diff
@@ -0,0 +1,218 @@
+///
+/// This function will return an error in the following situations:
+/// * The provided private key cannot be parsed as a valid `PrivateKeySigner` (returns `PostComputeInvalidEnclaveChallengePrivateKey`)
+/// * The signing operation fails (returns `PostComputeInvalidTeeSignature`)
+///
+/// # Example
+///
+/// ```
+/// let message_hash = "0x5cd0e9c5180dd35e2b8285d0db4ded193a9b4be6fbfab90cbadccecab130acad";
+/// let private_key = "0xdd3b993ec21c71c1f6d63a5240850e0d4d8dd83ff70d29e49247958548c1d479";
+///
+/// match sign_enclave_challenge(message_hash, private_key) {
+///     Ok(signature) => println!("Signature: {}", signature),
+///     Err(e) => eprintln!("Error: {:?}", e),
+/// }
+/// ```
+pub fn sign_enclave_challenge(
+    message_hash: &str,
+    enclave_challenge_private_key: &str,
+) -> Result<String, PostComputeError> {
+    let signer: PrivateKeySigner = enclave_challenge_private_key.parse::<PrivateKeySigner>()
+        .map_err(|_| PostComputeError::new(PostComputeInvalidEnclaveChallengePrivateKey))?;
+
+    let signature: Signature = signer.sign_message_sync(&hex_string_to_byte_array(&message_hash))
+        .map_err(|_| PostComputeError::new(PostComputeInvalidTeeSignature))?;
+
+    Ok(signature.to_string())
+}
+
+/// Generates a challenge signature for a given chain task ID.
+///
+/// This function retrieves the worker address and TEE challenge private key from the environment,
+/// then creates a message hash by concatenating and hashing the chain task ID and worker address.
+/// Finally, it signs this message hash with the private key.
+///
+/// # Arguments
+///
+/// * `chain_task_id` - A string identifier for the chain task
+///
+/// # Returns
+///
+/// * `Ok(String)` - The challenge signature as a hexadecimal string if successful
+/// * `Err(PostComputeError)` - An error if required environment variables are missing or if signing fails
+///
+/// # Errors
+///
+/// This function will return an error in the following situations:
+/// * The worker address environment variable is missing (returns `PostComputeWorkerAddressMissing`)
+/// * The TEE challenge private key environment variable is missing (returns `PostComputeTeeChallengePrivateKeyMissing`)
+/// * The signing operation fails (returns `PostComputeInvalidTeeSignature`)
+///
+/// # Environment Variables
+///
+/// * `SIGN_WORKER_ADDRESS` - The worker's address used in message hash calculation
+/// * `SIGN_TEE_CHALLENGE_PRIVATE_KEY` - The private key used for signing the challenge
+///
+/// # Example
+///
+/// ```
+/// // Assuming the necessary environment variables are set:
+/// // SIGN_WORKER_ADDRESS=0xabcdef123456789
+/// // SIGN_TEE_CHALLENGE_PRIVATE_KEY=0xdd3b993ec21c71c1f6d63a5240850e0d4d8dd83ff70d29e49247958548c1d479
+///
+/// let chain_task_id = "0x123456789abcdef";
+///
+/// match challenge(chain_task_id) {
+///     Ok(signature) => println!("Challenge signature: {}", signature),
+///     Err(e) => eprintln!("Error generating challenge: {:?}", e),
+/// }
+/// `
```

Comment:
Is there an error on this line ?
```suggestion
/// ```
```

Reply:
Resolved in [f39bf5d](https://github.com/iExecBlockchainComputing/tee-worker-post-compute-rust/pull/6/commits/f39bf5d901f0253e12ce5c1f43d946d8ef409488)